### PR TITLE
Add Open Voice Challenge option to the info page

### PR DESCRIPTION
**Pull request by rileyjshaw**
This commit updates privacy and mailing list options on the
/profile/info page.

Screen Shot 2019-10-15 at 5 13 26 PM

NOTE: As communicated in the TODOs, this commit is not ready for
production.

Test plan:

- Remove the `false && ...` condition from /web/src/components/pages/profile/info/info.tsx
- Visit localhost:9000/en/profile/info

The new option should appear, with new styles.

### CODE REVIEW

File: web/locales/en/messages.ftl [hunk 1]
```diff
@@ -8,6 +8,8 @@ email-input =
 submit-form-action = Submit
 loading = Loading…
 email-opt-in-info = I'd like to receive emails such as goal reminders, my progress updates and newsletters about Common Voice.
+email-opt-in-info-title = Join the Common Voice mailing list
+email-opt-in-info-sub = Receive emails such as goal reminders, progress updates and newsletters about Common Voice.
```
Comment (rileyjshaw) [inline web/locales/en/messages.ftl]: I'm still learning how the localization system works. As you can see, this is _very_ similar to the `email-opt-in-info` string above. When an edit like this is needed, if I update the initial value will changes eventually propagate to new locales? Or is it necessary to create a new ID like I did here?

File: web/src/components/pages/profile/info/info.css [hunk 2]
```diff
@@ -124,6 +132,12 @@
                 text-decoration: none;
             }
         }
+
+        a.terms {
```
Comment (rileyjshaw) [inline web/src/components/pages/profile/info/info.css]: I'm writing some pretty loosey-goosey CSS, so let me know if there's a style guide / ordering preference / etc that you were adhering to. One example that comes to mind is that I'm omitting leading `&`s where possible.

File: web/src/components/pages/profile/info/info.tsx [hunk 3]
```diff
@@ -323,25 +323,52 @@ function ProfilePage({
             </Tooltip>
 
             <div className="checkboxes">
+              {/* TODO(riley): Once Open Voice Challenge backend is set up,
+                               only show this to enrolled users. */}
+              {/* TODO(riley): Connect `onChange` and `checked` to the user
+                               model. */}
+              {false && (
```
Comment (rileyjshaw) [inline web/src/components/pages/profile/info/info.tsx]: 😅 I don't even want to put this behind `isProduction` yet...

File: web/src/components/pages/profile/info/info.tsx [hunk 4]
```diff
@@ -323,25 +323,52 @@ function ProfilePage({
...
+                    </Localized>
+                  </>
                 }
                 onChange={handleChangeFor('sendEmails')}
                 checked={sendEmails}
               />
 
               {!user.account && !isSubmitted && (
-                <React.Fragment>
+                <>
```
Comment (rileyjshaw) [inline web/src/components/pages/profile/info/info.tsx]: Any reason to use `React.Fragment` over `<>`?

Comment (Gregoor) [reply web/src/components/pages/profile/info/info.tsx]: I think the TS compiler needed a little longer to be ready for it, so no good reason not to use `<>` now :)

File: web/src/components/pages/profile/info/info.css (hunk 2 above)
Comment (Gregoor) [reply web/src/components/pages/profile/info/info.css]: I've been a loose goose myself, as you might've noticed. Had some ambitions about that a year ago, but that's about it in terms of style guide:
https://github.com/mozilla/voice-web/issues/912

File: web/locales/en/messages.ftl (hunk 1 above)
Comment (Gregoor) [reply web/locales/en/messages.ftl]: Unfortunately we do need new IDs when we want the localization community to notice. So you did the right thing!

File: web/src/components/pages/profile/info/info.css (hunk 2 above)
Comment (rileyjshaw) [reply web/src/components/pages/profile/info/info.css]: Oh cool, thanks for the link!

**Pull request merged**